### PR TITLE
Upgrade all deployments to apps/v1

### PR DESCRIPTION
**Pull request by nbkrause**
extensions/v1beta1 has been removed in kube 1.16

Signed-off-by: Noah Krause <krausenoah@gmail.com>

## Description
A few sentences describing the overall goals of the pull request's commits.

## Related Issues
List related issues.

## Testing
A few sentences describing what testing you've done, e.g., manual tests, automated tests, deployed in production, etc.

## Todos
- [ ] Tests
- [ ] Documentation

## Other
* If this is a documentation change, please open the pull request at https://github.com/datawire/ambassador-docs instead.
* Code changes should occur against `master`.

### CODE REVIEW

File: docs/reference/debugging.md [hunk 1]
```diff
@@ -306,7 +306,7 @@ If the generated Envoy configuration is not looking as expected, you can manuall
 
 ```shell
 $ kubectl scale deployment ambassador --replicas=1
-deployment.extensions "ambassador" scaled
```
Comment (kflynn) [inline docs/reference/debugging.md]: Just to doublecheck, this is correct?

File: docs/yaml/ambassador/ambassador-no-rbac.yaml [hunk 2]
```diff
@@ -30,7 +33,7 @@ spec:
     spec:
       containers:
       - name: ambassador
-        image: quay.io/datawire/ambassador:%version%
```
Comment (kflynn) [inline docs/yaml/ambassador/ambassador-no-rbac.yaml]: This definitely seems incorrect. 🙂

File: docs/reference/running.md [hunk 3]
```diff
@@ -18,12 +18,15 @@ Starting with Ambassador 0.35, we support running Ambassador as non-root. This i
...
 ```yaml
 ---
-apiVersion: extensions/v1beta1
+apiVersion: apps/v1
 kind: Deployment
 metadata:
   name: ambassador
 spec:
   replicas: 1
+  selector:
```
Comment (kflynn) [inline docs/reference/running.md]: There are a _lot_ of places in this PR where you add `selector`s to a spec. I think those should be a separate PR, unless I'm missing something.

Comment (nbkrause) [reply docs/reference/running.md]: The selector is required in the `apps/v1` version of `Deployment`

File: docs/yaml/ambassador/ambassador-no-rbac.yaml (hunk 2 above)
Comment (nbkrause) [reply docs/yaml/ambassador/ambassador-no-rbac.yaml]: This definitely is. Thanks for catching it

File: docs/reference/debugging.md (hunk 1 above)
Comment (nbkrause) [reply docs/reference/debugging.md]: Yes this is correct. Ambassador Deployments are no longer a `deployment.extension`. They are now a `deployment.apps`. It should be `deployment.apps/ambassador` however so I will quick make that change.

**Comment by nbkrause** [comment]
Since this is documentation only, will merge into `datawire/ambassador-docs`
